### PR TITLE
boards: esp32s2_devkitc: initial board support

### PR DESCRIPTION
**Pull request by wmrsouza**
Initial board files created to support the esp32s2_devkitc board.

### CODE REVIEW

**Comment by nordicjm** [comment]
Closing as duplicate of #66463

**Comment by sylvioalves** [comment]
@nordicjm, Sorry, it is not duplicated. #66463 is `esp32s3` SoC.

**Comment by nordicjm** [comment]
My bad, missed the 3 being different to 2 in branch name

File: boards/xtensa/esp32s2_devkitc/support/openocd.cfg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+set ESP_RTOS none
+set ESP32_ONLYCPU 1
+
+# Source the JTAG interface configuration file
+source [find interface/esp_usb_jtag.cfg]
```
Comment (marekmatej) [inline boards/xtensa/esp32s2_devkitc/support/openocd.cfg:5]: The ESP32-S2 doesn't have a built-in JTAG interface

**Review by marekmatej — CHANGES_REQUESTED** [review]
Except the openocd line, LGTM

**Review by nordicjm — CHANGES_REQUESTED** [review]
Needs to be migrated to hwmv2

**Comment by wmrsouza** [comment]
> Needs to be migrated to hwmv2

It was migrated to hwmv2

File: boards/xtensa/esp32s2_devkitc/support/openocd.cfg (hunk 1 above)
Comment (wmrsouza) [reply boards/xtensa/esp32s2_devkitc/support/openocd.cfg]: It was done

**Review by sylvioalves — APPROVED** [review]

**Review by LucasTambor — APPROVED** [review]

File: boards/espressif/esp32s2_devkitc/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+Building & Flashing
+*******************
+
+ESP-IDF bootloader
+==================
+
+The board is using the ESP-IDF bootloader as the default 2nd stage bootloader.
+It is build as a subproject at each application build. No further attention
+is expected from the user.
```
Comment (marekmatej) [inline boards/espressif/esp32s2_devkitc/doc/index.rst:96]: Can you please update this part to conform with the simple boot changes pending merge.

**Review by marekmatej — CHANGES_REQUESTED** [review]

**Review by nordicjm — APPROVED** [review]

File: boards/espressif/esp32s2_devkitc/doc/index.rst (hunk 2 above)
Comment (marekmatej) [reply boards/espressif/esp32s2_devkitc/doc/index.rst:96]: We can update it later.

**Review by marekmatej — APPROVED** [review]

**Pull request merged**